src/vcpkg/binarycaching.cpp

```diff
@@ -1290,6 +1304,190 @@ namespace
         std::vector<std::string> m_read_prefixes;
         std::vector<std::string> m_write_prefixes;
     };
+
+    bool coscli_stat(const VcpkgPaths& paths, const std::string& url)
```

Comment:
I would like to see the aws and cos paths merged such that the tool exe is passed in if the intent is that they truly have the same interface so that we don't forget to update one when changes are made to the other.

Reply:
#456 

I will merge gcs and aws first and then add oss and other object storage service later?